### ISSUE TITLE
error - The current model class (GPTNeoModel) is not compatible with `.generate()`, as it doesn't have a language model head. Please use one of the following classes instead: {'GPTNeoForCausalLM'}, for rik

### ISSUE THREAD

**Issue by rikbrown** [machaao/gpt-j-chatbot#9]
I'm getting this error when using locally:

```
error - The current model class (GPTNeoModel) is not compatible with `.generate()`, as it doesn't have a language model head. Please use one of the following classes instead: {'GPTNeoForCausalLM'}
```

**Comment by unixguru2k** [comment]
hi @rikbrown 

Can you also include the line number where this is happening?

**Comment by izardy** [comment]
> hi @rikbrown
> 
> Can you also include the line number where this is happening?

hu @unixguru2k , 

I'm getting the similar error message when using GPTNeo model . Below is the line number

"   1083 if generate_compatible_classes:
   1084     exception_message += f" Please use one of the following classes instead: {generate_compatible_classes}"
-> 1085 raise TypeError(exception_message)

"

**Comment by izardy** [comment]
anyway my attempt to mitigate the above is by implementing the following

```
from transformers import AutoTokenizer, AutoModelForCausalLM,pipeline
tokenizer = AutoTokenizer.from_pretrained("EleutherAI/gpt-neo-2.7B")
model = AutoModelForCausalLM.from_pretrained("EleutherAI/gpt-neo-2.7B")
generator = pipeline("text-generation", model=model,tokenizer=tokenizer)
```

**Comment by AytoMaximo** [comment]
Hi everyone! I've tried the fix from izardy above; however, I faced an error that my model and input_ids are located on different devices (CPU with CUDA conflict). I updated the line `model = AutoModelForCausalLM.from_pretrained("EleutherAI/gpt-neo-1.3B")` with ` .to(device)` at the end, but unfortunately, using CUDA as the device never returns an answer. It's busy at 100%, but with no result. If I force CPU usage, it will come up with something in 1 minute. I'm a newbie in ML stuff, but what can this issue be about? Is it relative to izardy's workaround at all?
